pp.c

```diff
@@ -6033,30 +6034,10 @@ PP(pp_split)
             oldsave = PL_savestack_ix;
         }
 
-	realarray = 1;
-	PUTBACK;
-	av_extend(ary,0);
-	(void)sv_2mortal(SvREFCNT_inc_simple_NN(sv));
```

Comment:
The inc + mortal should be kept to ensure ary is still live when we get around to pushing to it.

Reply:
Thanks for looking at this. Is the mortal for the ary or for the string being split? 

I had rationalized it as being the string sv, which was incremented in case it was on the ary, as we wouldn't want the string sv to be destroyed by the ary_clear prior to the split operations happening on it.

Reply:
You're right, I misread that code.

I mean ary, since code in the regexp might release it.

Reply:
So a new `(void)sv_2mortal(SvREFCNT_inc_simple_NN(ary));` in roughly the same place near `realarray = 1;`?
(I'm happy to put that in, just checking that I understand you correctly.)

Reply:
The PR as-is already survives `perl -e '@ary = split(/\w(?{ undef @ary })/, "abc");'`. How can I test that an added `(void)sv_2mortal(SvREFCNT_inc_simple_NN(ary));` works correctly? (i.e. What should be in the regex to get it to release _ary_?)

Reply:
The problem isn't modifications to the array itself, but the array itself being destroyed `perl -e '@ary = split(/\w(?{ *ary = 0 })/, "abc");'` or `perl -e '$ary = []; @$ary = split(/\w(?{ $ary = [] })/, "abc");'`.

It's not directly related to what you're trying to fix though, it's a partial fix for stack-not-refcounted.

Reply:
Ah, I see, thanks Tony. I've pushed an extra commit. Please let me know if any other changes are required.